### PR TITLE
PI-14967: [VRF] Create controller + basic entities on SDK 

### PR DESCRIPTION
**Pull request by dmarques10**
JIRA ticket:

- [x] Approved by a PI tech lead

### CODE REVIEW

File: src/main/java/com/appdirect/sdk/vendorrequiredfields/model/Level.java [hunk 1]
```diff
@@ -0,0 +1,5 @@
+package com.appdirect.sdk.vendorrequiredfields.model;
+
+public enum Level {
```
Comment (esteban-mallen) [inline src/main/java/com/appdirect/sdk/vendorrequiredfields/model/Level.java]: Did we agree on this name?

Comment (esteban-mallen) [reply src/main/java/com/appdirect/sdk/vendorrequiredfields/model/Level.java]: Also, this enum is not being used anywhere right now, should it be present in `VendorRequiredField`?

File: src/main/java/com/appdirect/sdk/vendorrequiredfields/model/FlowType.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+	END_USER_FLOW("END_USER_FLOW"), RESELLER_FLOW("RESELLER_FLOW");
+
+	private String value;
+
+	private FlowType(String value) {
+		this.value = value;
+	}
+
+	public static FlowType fromValue(String value) {
+		for (FlowType flowType : values()) {
```
Comment (esteban-mallen) [inline src/main/java/com/appdirect/sdk/vendorrequiredfields/model/FlowType.java]: Or using Streams:
```suggestion
		return Stream.of(values())
			.filter(flowType -> flowType.value.equalsIgnoreCase(value))
			.findFirst()
			.orElseThrow(() -> new IllegalArgumentException(
				"Unknown enum type " + value + ", Allowed values are " + Arrays.toString(values())));
```

File: src/main/java/com/appdirect/sdk/vendorrequiredfields/model/VendorRequiredFieldResponse.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+@Setter
+@ToString
+@EqualsAndHashCode
+@AllArgsConstructor
+@JsonInclude(JsonInclude.Include.NON_EMPTY)
+public class VendorRequiredFieldResponse {
+
+	private List<Form> forms;
+
+	private int order;
```
Comment (gastontulipani) [inline src/main/java/com/appdirect/sdk/vendorrequiredfields/model/VendorRequiredFieldResponse.java]: Why do we need an order here? We need it in the Form or the Field

File: src/main/java/com/appdirect/sdk/vendorrequiredfields/VendorRequiredFieldController.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+ */
+@RestController
+@RequiredArgsConstructor
+public class VendorRequiredFieldController {
+	private final VendorRequiredFieldHandler vendorRequiredFieldHandler;
+
+	@RequestMapping(method = GET,
+		value = "/api/v1/admin/requiredFields",
+		produces = APPLICATION_JSON_VALUE,
+		consumes = APPLICATION_JSON_VALUE)
```
Comment (gastontulipani) [inline src/main/java/com/appdirect/sdk/vendorrequiredfields/VendorRequiredFieldController.java]: Why are we expecting a json?

File: src/main/java/com/appdirect/sdk/vendorrequiredfields/VendorRequiredFieldConfiguration.java [hunk 5]
```diff
@@ -0,0 +1,24 @@
+package com.appdirect.sdk.vendorrequiredfields;
+
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+import com.appdirect.sdk.vendorrequiredfields.model.FlowType;
+import com.appdirect.sdk.vendorrequiredfields.model.OperationType;
+
+
```
Comment (gastontulipani) [inline src/main/java/com/appdirect/sdk/vendorrequiredfields/VendorRequiredFieldConfiguration.java]: Shouldn't we remove this line?

**Pull request merged**